File: BCC221/semana06-tp01/Apartamento.cpp
#include "Apartamento.h"

Apartamento :: Apartamento(const int id, const float valor, const string proprietario,
                        const string rua, const string bairro, const string cidade, const int numero,
                        const int quartos, const int banheiros,const int andar, const float taxa_condominio,
                        const bool elevador, const bool sacada) : Imovel(id, valor, proprietario, rua, bairro,
                        cidade, numero, quartos, banheiros), andar(andar), taxa_condominio(taxa_condominio),
                        elevador(elevador), sacada(sacada){}

int Apartamento::getAndar() const{
    return andar;
}

void Apartamento::setAndar(int andar){
    this->andar = andar;
}

float Apartamento::getTaxaCondominio() const{
    return taxa_condominio;
}

void Apartamento::setTaxaCondominio(float taxa_condominio){
    this->taxa_condominio = taxa_condominio;
}

bool Apartamento::getElevador() const{
    return elevador;
}

void Apartamento::setElevador(bool elevador){
    this->elevador = elevador;
}

bool Apartamento::getSacada() const{
    return sacada;
}

void Apartamento::setSacada(bool sacada){
    this->sacada = sacada;
}

void Apartamento::saida(ostream& out){
    Imovel::saida(out);
    if (this->elevador){
        out << "    Elevador: SIM" << endl;
    }
    else{
        out << "    Elevador: NAO" << endl;
    }
}
    
ostream& operator<<(ostream& out, Apartamento& obj){
    obj.saida(out);
    return out;
}

void Apartamento::imprimirTudo(){
    cout << "    Andar: " << (this->getAndar());
    cout << "\n    Taxa de condominio: " << (this->getTaxaCondominio());
    cout << "\n    Sacada: " << (this->getSacada() ? "SIM" : "NAO") << endl;
}

Apartamento::~Apartamento() = default;

File: BCC221/semana14-tp02/src/main/java/backend/Apartamento.java
package backend;

public class Apartamento extends Imovel {

    private int andar;
    private float taxa_condominio;
    private boolean elevador;
    private boolean sacada;
    
    //Construtor geral
    public Apartamento(double valor, String proprietario, String rua, String bairro, String cidade,
            int numero, int quartos, int banheiros, int andar, float taxa_condominio, boolean elevador,
            boolean sacada) {
        super(valor, proprietario, rua, bairro, cidade, numero, quartos, banheiros);
        this.andar = andar;
        this.taxa_condominio = taxa_condominio;
        this.elevador = elevador;
        this.sacada = sacada;
    }

    //Construtor utilizando imovel e demais atributos
    public Apartamento(Imovel imovel, int andar, float taxa_condominio, boolean elevador, boolean sacada) {
        super(imovel);
        this.andar = andar;
        this.taxa_condominio = taxa_condominio;
        this.elevador = elevador;
        this.sacada = sacada;
    }

    //Construtor inicializando padrões 
    public Apartamento() {
        super(0.0, "", "", "", "", 0, 0, 0);
        this.andar = 0;
        this.taxa_condominio = 0;
        this.elevador = false;
        this.sacada = false;
    }

    public int getAndar() {
        return andar;
    }

    public void setAndar(int andar) {
        if (andar > 0) {
            this.andar = andar;
        }
    }

    public float getTaxa_condominio() {
        return taxa_condominio;
    }

    public void setTaxa_condominio(float taxa_condominio) {
        if (taxa_condominio > 0) {
            this.taxa_condominio = taxa_condominio;
        }
    }

    public boolean isElevador() {
        return elevador;
    }

    public void setElevador(boolean elevador) {
        this.elevador = elevador;
    }

    public boolean isSacada() {
        return sacada;
    }

    public void setSacada(boolean sacada) {
        this.sacada = sacada;
    }

    //Sobreescrita do método toString
    @Override
    public String toString() {
        return super.toString() + "Elevador: " + (this.elevador ? "SIM" : "NAO");
    }

    //Sobreescrita do método que imprime atributos específicos da classe
    @Override
    public String imprimirClasse() {
        return super.toString() + "\n\tCaracterística Especifica\n"
                + "\t\tAndar: " + this.andar
                + "\n\t\tTaxa de condominio: " + this.taxa_condominio
                + "\n\t\tElevador: " + (this.elevador ? "SIM" : "NAO")
                + "\n\t\tSacada: " + (this.sacada ? "SIM" : "NAO");
    }
}
